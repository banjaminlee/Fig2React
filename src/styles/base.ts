import { MediaQueries, Breakpoints } from './types';

export const breakpointMap: { [key: string]: number } = {
  xs: 370,
  sm: 576,
  md: 768,
  lg: 1024,
  xl: 1200,
  xxl: 1500,
};

export const breakpoints: Breakpoints = Object.values(breakpointMap).map(
  (breakpoint) => `${breakpoint}px`,
);

export const mediaQueries: MediaQueries = {
  xs: `@media screen and (min-width: ${breakpointMap.xs}px)`,
  sm: `@media screen and (min-width: ${breakpointMap.sm}px)`,
  md: `@media screen and (min-width: ${breakpointMap.md}px)`,
  lg: `@media screen and (min-width: ${breakpointMap.lg}px)`,
  xl: `@media screen and (min-width: ${breakpointMap.xl}px)`,
  xxl: `@media screen and (min-width: ${breakpointMap.xxl}px)`,
};
