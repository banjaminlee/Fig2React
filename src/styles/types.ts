export type Breakpoints = string[];

export type MediaQueries = {
  xs: string;
  sm: string;
  md: string;
  lg: string;
  xl: string;
  xxl: string;
};

export type Colors = {
  periwinkle: string;
  periwinkle_light: string;
  sand: string;
  sand_dark: string;
  clay: string;
  clay_dark: string;
  sky: string;
  sky_dark: string;
  black: string;
  black_light: string;
  background_primary: string;
  background_secondary: string;
  background_tertiary: string;
  background_progress: string;
  text_primary: string;
  text_secondary: string;
  text_bold: string;
  icon_primary: string;
  carousel_dot1: string;
  carousel_dot2: string;
  carousel_dot3: string;
  selected_toggle_background: string;
  selected_secondary_toggle_background: string;
  primary_toggle: string;
  secondary_toggle: string;
  white: string;
  grays: string;
  warning: string;
  info: string;
  error: string;
  success: string;
  ui_primary: string;
  ui_secondary: string;
  ui_tertiary: string;
  ui_inverted: string;
  ui_highlight_primary: string;
  ui_highlight_secondary: string;
  ui_highlight_tertiary: string;
  ui_divider: string;
  ui_error: string;
  ui_success: string;
  ui_pending: string;
  ui_link: string;
  ui_background: string;
  ui_text: string;
  ui_rich_text_editor_preview_box: string;
};

export type Shadows = {
  shadow00: string;
  shadow01: string;
  shadow02: string;
  shadow03: string;
  shadow04: string;
  shadow06: string;
  shadow08: string;
  shadow09: string;
  shadow12: string;
  shadow16: string;
  shadow24: string;
};

export type BaseFonts = {
  fontFamily?: string;
  fontWeight?: number;
  fontSize?: string;
  lineHeight?: string;
  letterSpacing?: string;
  fontStyle?: string;
  textDecoration?: string;
  textTransform?: string;
};
