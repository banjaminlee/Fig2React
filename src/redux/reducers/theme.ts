import { SET_THEME } from '../actions/theme-actions';

import { lightTheme, darkTheme } from '../../styles/theme';

// Define your state here
const initialState = {
  loading: false,
  theme: lightTheme,
  themeName: 'light',
  themes: {
    dark: darkTheme,
    light: lightTheme,
  },
};

// This export default will control your state for your application
export default (state = initialState, { type, payload }: any) => {
  switch (type) {
    case SET_THEME:
      return {
        ...state,
        theme: payload.theme,
        themeName: payload.name,
      };
    // Return default state if you didn't match any case
    default:
      return state;
  }
};
