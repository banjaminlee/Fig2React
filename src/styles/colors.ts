import { Colors } from './types';
const palette = {
  white: '#fff',
  black: '#000',
  periwinkle: {
    50: '#eaecff',
    100: '#c9ceff',
    200: '#a4aeff',
    300: '#7b8dff',
    400: '#5870ff',
    500: '#3055fc',
    600: '#2b4cfc',
    700: '#1d40e3',
    800: '#0c34d8',
    900: '#011dbf',
    1000: '#7594FF',
  },
  sand: {
    50: '#fcf7ed',
    100: '#feeac5',
    200: '#ffdc97',
    300: '#ffcd68',
    400: '#ffc147',
    500: '#ffb533',
    600: '#ffa92e',
    700: '#ff9929',
    800: '#fb8a24',
    900: '#f2721c',
    1000: '#F3EEE4',
  },
  clay: {
    10: '#C2C2C2',
    20: '#D9D9D9',
    50: '#f9e8e6',
    100: '#facbb8',
    200: '#f7a98b',
    300: '#f3895d',
    400: '#f17038',
    500: '#ef5910',
    600: '#e4530c',
    700: '#d74c06',
    800: '#c94503',
    900: '#b03800',
    1000: '#B73D01',
  },
  sky: {
    50: '#d7f5ff',
    100: '#a1e3fa',
    200: '#62cff4',
    300: '#00bcee',
    400: '#00aeec',
    500: '#00a0e9',
    600: '#0092db',
    700: '#0080c9',
    800: '#006fb6',
    900: '#004f95',
    1000: '#8FEAFE',
  },
  gray: {
    50: '#f6f6f6',
    100: '#eeeeee',
    200: '#e2e2e2',
    300: '#cbcbcb',
    400: '#afafaf',
    500: '#6b6b6b',
    600: '#545454',
    700: '#333333',
    800: '#1f1f1f',
    900: '#141414',
  },
  green: {
    100: '#addec9',
    200: '#66D19e',
    300: '#06c167',
    400: '#4CAF50',
    500: '#03703c',
    600: '#545454',
    700: '#333333',
  },
  yellow: {
    100: '#fff2d9',
    200: '#ffe3ac',
    300: '#ffcf70',
    400: '#FFC043',
    500: '#bc8b2c',
    600: '#996f00',
    700: '#674d1b',
  },
  red: {
    100: '#fed7d2',
    200: '#f1998e',
    300: '#e85c4a',
    400: '#e11900',
    500: '#ab1300',
    600: '#870f00',
    700: '#5a0a00',
  },
  brown: {
    100: '#c94403',
  },
};

export const lightColors: Colors = {
  periwinkle: palette.periwinkle[400],
  periwinkle_light: palette.periwinkle[1000],
  sand: palette.sand[50],
  sand_dark: palette.sand[1000],
  clay: palette.clay[800],
  clay_dark: palette.clay[1000],
  sky: palette.sky[50],
  sky_dark: palette.sky[100],
  black: palette.black,
  black_light: palette.gray[700],
  background_primary: palette.white,
  background_secondary: palette.sand[50],
  background_tertiary: palette.sand[1000],
  text_primary: palette.gray[800],
  text_secondary: palette.gray[600],
  text_bold: palette.black,
  icon_primary: palette.clay[800],
  carousel_dot1: palette.black,
  carousel_dot2: palette.gray[700],
  carousel_dot3: palette.sand[1000],
  selected_toggle_background: palette.brown[100],
  selected_secondary_toggle_background: palette.sky[1000],
  primary_toggle: palette.clay[10],
  secondary_toggle: palette.clay[20],
  white: palette.white,
  grays: palette.gray[300],
  background_progress: palette.gray[200],
  warning: palette.yellow[400],
  info: palette.periwinkle[400],
  error: palette.red[400],
  success: palette.green[400],
  ui_primary: palette.periwinkle[400],
  ui_secondary: palette.sand[50],
  ui_tertiary: palette.clay[800],
  ui_inverted: palette.clay[800],
  ui_highlight_primary: palette.clay[800],
  ui_highlight_secondary: palette.white,
  ui_highlight_tertiary: palette.clay[800],
  ui_divider: palette.gray[500],
  ui_error: palette.red[400],
  ui_success: palette.green[400],
  ui_pending: palette.yellow[400],
  ui_link: palette.clay[1000],
  ui_background: palette.black,
  ui_text: palette.white,
  ui_rich_text_editor_preview_box: palette.yellow[100],
};

export const darkColors: Colors = {
  periwinkle: palette.periwinkle[400],
  periwinkle_light: palette.periwinkle[1000],
  sand: palette.sand[50],
  sand_dark: palette.sand[1000],
  clay: palette.clay[800],
  clay_dark: palette.clay[1000],
  sky: palette.sky[50],
  sky_dark: palette.sky[100],
  black: palette.black,
  black_light: palette.gray[700],
  background_primary: palette.black,
  background_secondary: palette.gray[700],
  background_tertiary: palette.sand[1000],
  background_progress: palette.gray[200],
  text_primary: palette.white,
  text_secondary: palette.sky[50],
  text_bold: palette.periwinkle[400],
  icon_primary: palette.periwinkle[400],
  carousel_dot1: palette.gray[50],
  carousel_dot2: palette.clay[800],
  carousel_dot3: palette.white,
  selected_toggle_background: palette.brown[100],
  selected_secondary_toggle_background: palette.sky[1000],
  primary_toggle: palette.clay[10],
  secondary_toggle: palette.clay[20],
  white: palette.white,
  grays: palette.gray[500],
  warning: palette.yellow[500],
  info: palette.periwinkle[400],
  error: palette.red[500],
  success: palette.green[500],
  ui_primary: palette.periwinkle[400],
  ui_secondary: palette.sand[50],
  ui_tertiary: palette.clay[800],
  ui_inverted: palette.clay[800],
  ui_highlight_primary: palette.clay[800],
  ui_highlight_secondary: palette.white,
  ui_highlight_tertiary: palette.clay[800],
  ui_divider: palette.gray[500],
  ui_error: palette.red[500],
  ui_success: palette.green[500],
  ui_pending: palette.yellow[500],
  ui_link: palette.clay[800],
  ui_background: palette.clay[800],
  ui_text: palette.white,
  ui_rich_text_editor_preview_box: palette.yellow[100],
};
