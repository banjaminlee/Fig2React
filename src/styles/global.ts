import { createGlobalStyle } from 'styled-components';
import globaltokenCSS from '../tokens.css';

export default createGlobalStyle`

  ${globaltokenCSS}
  
  html, body {
    margin: 0;
    padding: 0;
    outline: 0;
    box-sizing: border-box;
    line-height: 1.25rem;
    font-weight: 400;
    font-family: 'Roc Grotesk Regular';
    font-size: calc([minimum size] + ([maximum size] - [minimum size]) * ((100vw - [minimum viewport width]) / ([maximum viewport width] - [minimum viewport width]))); 
    background: ${({ theme }) => theme.colors.background_primary};
    color: ${({ theme }) => theme.colors.text_primary};
  }

  /* width */
  ::-webkit-scrollbar {
    width: 8px;
    height: 8px;
  }

  /* Track */
  ::-webkit-scrollbar-track {
    background: transparent;
    border-radius: 36px;
  }

  /* Handle */
  ::-webkit-scrollbar-thumb {
    background: ${({ theme }) => theme.colors.text_primary};
    border-radius: 36px;
  }
`;
