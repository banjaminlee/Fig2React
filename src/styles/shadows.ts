import { Shadows } from './types';

export const lightShadows: Shadows = {
  shadow00: 'none',
  shadow01:
    '0px 1px 1px rgba(0, 0, 0, 0.14), 0px 2px 1px rgba(0, 0, 0, 0.12), 0px 1px 3px rgba(0, 0, 0, 0.2)',
  shadow02:
    '0px 2px 2px rgba(0, 0, 0, 0.14), 0px 3px 1px rgba(0, 0, 0, 0.12), 0px 1px 5px rgba(0, 0, 0, 0.2)',
  shadow03:
    '0px 3px 4px rgba(0, 0, 0, 0.14), 0px 3px 3px rgba(0, 0, 0, 0.12), 0px 1px 8px rgba(0, 0, 0, 0.2)',
  shadow04:
    '0px 4px 5px rgba(0, 0, 0, 0.14), 0px 1px 10px rgba(0, 0, 0, 0.12), 0px 2px 4px rgba(0, 0, 0, 0.2)',
  shadow06:
    '0px 6px 10px rgba(0, 0, 0, 0.14), 0px 1px 18px rgba(0, 0, 0, 0.12), 0px 3px 5px rgba(0, 0, 0, 0.2)',
  shadow08:
    '0px 8px 10px rgba(0, 0, 0, 0.14), 0px 3px 14px rgba(0, 0, 0, 0.12), 0px 5px 5px rgba(0, 0, 0, 0.2)',
  shadow09:
    '0px 9px 12px rgba(0, 0, 0, 0.14), 0px 3px 16px rgba(0, 0, 0, 0.12), 0px 5px 6px rgba(0, 0, 0, 0.2)',
  shadow12:
    '0px 12px 17px rgba(0, 0, 0, 0.14), 0px 5px 22px rgba(0, 0, 0, 0.12), 0px 7px 8px rgba(0, 0, 0, 0.2)',
  shadow16:
    '0px 16px 24px rgba(0, 0, 0, 0.14), 0px 6px 30px rgba(0, 0, 0, 0.12), 0px 8px 10px rgba(0, 0, 0, 0.2)',
  shadow24:
    '0px 24px 38px rgba(0, 0, 0, 0.14), 0px 9px 46px rgba(0, 0, 0, 0.12), 0px 11px 15px rgba(0, 0, 0, 0.2)',
};

export const darkShadows: Shadows = {
  shadow00: 'none',
  shadow01:
    '0px 1px 1px rgba(0, 0, 0, 0.14), 0px 2px 1px rgba(0, 0, 0, 0.12), 0px 1px 3px rgba(0, 0, 0, 0.2)',
  shadow02:
    '0px 2px 2px rgba(0, 0, 0, 0.14), 0px 3px 1px rgba(0, 0, 0, 0.12), 0px 1px 5px rgba(0, 0, 0, 0.2)',
  shadow03:
    '0px 3px 4px rgba(0, 0, 0, 0.14), 0px 3px 3px rgba(0, 0, 0, 0.12), 0px 1px 8px rgba(0, 0, 0, 0.2)',
  shadow04:
    '0px 4px 5px rgba(0, 0, 0, 0.14), 0px 1px 10px rgba(0, 0, 0, 0.12), 0px 2px 4px rgba(0, 0, 0, 0.2)',
  shadow06:
    '0px 6px 10px rgba(0, 0, 0, 0.14), 0px 1px 18px rgba(0, 0, 0, 0.12), 0px 3px 5px rgba(0, 0, 0, 0.2)',
  shadow08:
    '0px 8px 10px rgba(0, 0, 0, 0.14), 0px 3px 14px rgba(0, 0, 0, 0.12), 0px 5px 5px rgba(0, 0, 0, 0.2)',
  shadow09:
    '0px 9px 12px rgba(0, 0, 0, 0.14), 0px 3px 16px rgba(0, 0, 0, 0.12), 0px 5px 6px rgba(0, 0, 0, 0.2)',
  shadow12:
    '0px 12px 17px rgba(0, 0, 0, 0.14), 0px 5px 22px rgba(0, 0, 0, 0.12), 0px 7px 8px rgba(0, 0, 0, 0.2)',
  shadow16:
    '0px 16px 24px rgba(0, 0, 0, 0.14), 0px 6px 30px rgba(0, 0, 0, 0.12), 0px 8px 10px rgba(0, 0, 0, 0.2)',
  shadow24:
    '0px 24px 38px rgba(0, 0, 0, 0.14), 0px 9px 46px rgba(0, 0, 0, 0.12), 0px 11px 15px rgba(0, 0, 0, 0.2)',
};
