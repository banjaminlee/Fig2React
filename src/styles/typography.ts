import { BaseFonts } from './types';

// New
const display_md = {
  fontFamily: 'Roc Grotesk Extra Bold',
  fontWeight: 100,
  fontSize: '5.25rem',
  lineHeight: '5.313rem',
  letterSpacing: '0rem',
};

const heading_xxl = {
  fontFamily: 'Roc Grotesk Bold',
  fontWeight: 100,
  fontSize: '4rem',
  lineHeight: '4.625rem',
  letterSpacing: '-0.063rem',
};

const heading_xl = {
  fontFamily: 'Roc Grotesk Bold',
  fontWeight: 100,
  fontSize: '3rem',
  lineHeight: '3.625rem',
  letterSpacing: '0rem',
};

const heading_lg = {
  fontFamily: 'Roc Grotesk Bold',
  fontWeight: 100,
  fontSize: '2.125rem',
  lineHeight: '2.75rem',
  letterSpacing: '0rem',
};

const heading_md = {
  fontFamily: 'Roc Grotesk Bold',
  fontWeight: 100,
  fontSize: '1.5rem',
  lineHeight: '2.125rem',
  letterSpacing: '0rem',
};
const heading_sm = {
  fontFamily: 'Roc Grotesk Bold',
  fontWeight: 100,
  fontSize: '1.25rem',
  lineHeight: '1.875rem',
  letterSpacing: '0rem',
};

const heading_xl_med = {
  fontFamily: 'Roc Grotesk Medium',
  fontWeight: 100,
  fontSize: '3rem',
  lineHeight: '3.625rem',
  letterSpacing: '0rem',
};

const heading_lg_med = {
  fontFamily: 'Roc Grotesk Medium',
  fontWeight: 100,
  fontSize: '2.125rem',
  lineHeight: '2.75rem',
  letterSpacing: '0rem',
};

const heading_md_med = {
  fontFamily: 'Roc Grotesk Medium',
  fontWeight: 100,
  fontSize: '1.5rem',
  lineHeight: '2.125rem',
  letterSpacing: '0rem',
};
const heading_sm_med = {
  fontFamily: 'Roc Grotesk Medium',
  fontWeight: 100,
  fontSize: '1.25rem',
  lineHeight: '2.125rem',
  letterSpacing: '0rem',
};
const body_lg = {
  fontFamily: 'Roc Grotesk Regular',
  fontWeight: 100,
  fontSize: '1.125rem',
  lineHeight: '1.75rem',
  letterSpacing: '0rem',
};
const body_md = {
  fontFamily: 'Roc Grotesk Regular',
  fontWeight: 100,
  fontSize: '1rem',
  lineHeight: '1.25rem',
  letterSpacing: '0rem',
};
const body_sm = {
  fontFamily: 'Roc Grotesk Regular',
  fontWeight: 100,
  fontSize: '0.75rem',
  lineHeight: '1rem',
  letterSpacing: '0rem',
};
const body_lg_bold = {
  fontFamily: 'Roc Grotesk Bold',
  fontWeight: 100,
  fontSize: '1.125rem',
  lineHeight: '1.75rem',
  letterSpacing: '0rem',
};
const body_md_bold = {
  fontFamily: 'Roc Grotesk Bold',
  fontWeight: 100,
  fontSize: '1rem',
  lineHeight: '1.25rem',
  letterSpacing: '0rem',
};
const body_sm_bold = {
  fontFamily: 'Roc Grotesk Bold',
  fontWeight: 100,
  fontSize: '0.75rem',
  lineHeight: '1rem',
  letterSpacing: '0rem',
};

const caption_md = {
  fontFamily: 'Roc Grotesk Regular',
  fontWeight: 100,
  fontSize: '0.625rem',
  lineHeight: '0.938rem',
  letterSpacing: '0rem',
};
const caption_md_bold = {
  fontFamily: 'Roc Grotesk Bold',
  fontWeight: 100,
  fontSize: '1rem',
  lineHeight: '0.938rem',
  letterSpacing: '0rem',
};
const button_md = {
  fontFamily: 'Roc Grotesk Medium',
  fontWeight: 100,
  fontSize: '0.875rem',
  lineHeight: '1.625rem',
  letterSpacing: '0rem',
};

const eyebrow_md = {
  fontFamily: 'Roc Grotesk Bold',
  fontWeight: 100,
  fontSize: '0.875rem',
  lineHeight: '1.5rem',
  letterSpacing: '0rem',
};
const text_cta = {
  fontFamily: 'Roc Grotesk Bold',
  fontWeight: 100,
  letterSpacing: '0rem',
  fontSize: '0.875rem',
  lineHeight: '1rem',
};
const heading_mobile_xl = {
  fontFamily: 'Roc Grotesk Extra Bold',
  fontWeight: 100,
  letterSpacing: '0rem',
  fontSize: '3rem',
  lineHeight: '3.625rem',
};
const heading_mobile_lg = {
  fontFamily: 'Roc Grotesk Extra Bold',
  fontWeight: 100,
  letterSpacing: '0rem',
  fontSize: '2.5rem',
  lineHeight: '3.125rem',
};
const price_cut = {
  fontFamily: 'Roc Grotesk Extra Bold',
  fontWeight: 700,
  letterSpacing: '0rem',
  fontSize: '1.125rem',
  lineHeight: '1.5rem',
  textDecorationLine: 'line-through',
  textDecorationStyle: 'solid',
};
const RocGroteskBlack = {
  fontFamily: 'Roc Grotesk Black',
  fontSize: '1rem',
  lineHeight: '1.25rem',
  letterSpacing: '0rem',
  fontWeight: 100,
};
const RocGroteskBold = {
  fontFamily: 'Roc Grotesk Bold',
  fontSize: '1rem',
  lineHeight: '1.25rem',
  letterSpacing: '0rem',
  fontWeight: 100,
};
const RocGroteskExtraBold = {
  fontFamily: 'Roc Grotesk Extra Bold',
  fontSize: '1rem',
  lineHeight: '1.25rem',
  letterSpacing: '0rem',
  fontWeight: 100,
};
const RocGroteskExtraLight = {
  fontFamily: 'Roc Grotesk Extra Light',
  fontSize: '1rem',
  lineHeight: '1.25rem',
  letterSpacing: '0rem',
  fontWeight: 100,
};
const RocGroteskHeavy = {
  fontFamily: 'Roc Grotesk Heavy',
  fontSize: '1rem',
  lineHeight: '1.25rem',
  letterSpacing: '0rem',
  fontWeight: 100,
};
const RocGroteskLight = {
  fontFamily: 'Roc Grotesk Light',
  fontSize: '1rem',
  lineHeight: '1.25rem',
  letterSpacing: '0rem',
  fontWeight: 100,
};
const RocGroteskMedium = {
  fontFamily: 'Roc Grotesk Medium',
  fontSize: '1rem',
  lineHeight: '1.25rem',
  letterSpacing: '0rem',
  fontWeight: 100,
};
const RocGroteskRegular = {
  fontFamily: 'Roc Grotesk Regular',
  fontSize: '1rem',
  lineHeight: '1.25rem',
  letterSpacing: '0rem',
  fontWeight: 100,
};
const RocGroteskThin = {
  fontFamily: 'Roc Grotesk Thin',
  fontSize: '1rem',
  lineHeight: '1.25rem',
  letterSpacing: '0rem',
  fontWeight: 100,
};
let typography: {
  [key: string]: BaseFonts;
} = {};

typography = {
  display_md,
  heading_xxl,
  heading_xl,
  heading_lg,
  heading_md,
  heading_sm,
  RocGroteskBlack,
  RocGroteskBold,
  RocGroteskExtraBold,
  RocGroteskExtraLight,
  RocGroteskHeavy,
  RocGroteskLight,
  RocGroteskMedium,
  RocGroteskRegular,
  RocGroteskThin,
  heading_xl_med,
  heading_lg_med,
  heading_md_med,
  heading_sm_med,
  body_lg,
  body_md,
  body_sm,
  body_lg_bold,
  body_md_bold,
  body_sm_bold,
  caption_md,
  button_md,
  eyebrow_md,
  text_cta,
  heading_mobile_xl,
  heading_mobile_lg,
  price_cut,
  caption_md_bold,
};

export default typography;
