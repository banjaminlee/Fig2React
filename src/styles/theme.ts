import { DefaultTheme } from 'styled-components';
import { mediaQueries } from './base';
import { lightColors, darkColors } from './colors';
import { lightShadows, darkShadows } from './shadows';

export const lightTheme: DefaultTheme = {
  colors: lightColors,
  mediaQueries,
  shadows: lightShadows,
};

export const darkTheme: DefaultTheme = {
  colors: darkColors,
  mediaQueries,
  shadows: darkShadows,
};
