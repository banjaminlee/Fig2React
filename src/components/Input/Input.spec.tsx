import React from 'react';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import { Input, InputByRef, PhoneInput, EmailInput, PasswordInput } from './index';

import { ThemeProvider } from 'styled-components';

import { lightTheme } from 'styles/theme';

describe('Input', () => {
  const defaultProps = {
    id: 'input-1',
    name: 'input-1',
    label: 'Input 1',
    value: '',
    placeholder: 'Enter text',
    cbHandler: jest.fn(),
  };

  it('renders the input with the correct label', () => {
    const { getByLabelText } = render(
      <ThemeProvider theme={lightTheme}>
        <Input {...defaultProps} />
      </ThemeProvider>,
    );
    const inputElement = waitFor(() => getByLabelText('Input 1'));
    waitFor(() => {
      expect(inputElement).toBeInTheDocument();
    });
  });

  it('calls the onChange callback handler on input change', () => {
    const { getByLabelText } = render(
      <ThemeProvider theme={lightTheme}>
        <Input {...defaultProps} />
      </ThemeProvider>,
    );

    waitFor(() => fireEvent.change(getByLabelText('Input 1'), { target: { value: 'hello' } }));
    waitFor(() => expect(defaultProps.cbHandler).toHaveBeenCalled());
  });

  it('displays the error message when there is an error', () => {
    const { getByText } = render(
      <ThemeProvider theme={lightTheme}>
        <Input {...defaultProps} />
      </ThemeProvider>,
    );

    waitFor(() => expect(getByText('Invalid input')).toBeInTheDocument());
  });

  it('does not display the error message when there is no error', () => {
    const { queryByText } = render(
      <ThemeProvider theme={lightTheme}>
        <Input {...defaultProps} />
      </ThemeProvider>,
    );
    const errorElement = queryByText('Invalid input');
    waitFor(() => expect(errorElement).not.toBeInTheDocument());
  });

  it('applies the "disabled" attribute when the input is disabled', () => {
    const { getByLabelText } = render(
      <ThemeProvider theme={lightTheme}>
        <Input {...defaultProps} disabled={true} />
      </ThemeProvider>,
    );

    waitFor(() => expect(getByLabelText('Input 1')).toBeDisabled());
  });

  it('does not apply the "disabled" attribute when the input is enabled', () => {
    const { getByLabelText } = render(
      <ThemeProvider theme={lightTheme}>
        <Input {...defaultProps} />
      </ThemeProvider>,
    );
    waitFor(() => {
      expect(getByLabelText('Input 1')).not.toBeDisabled();
    });
  });
});

describe('InputByRef', () => {
  it('should render correctly', () => {
    const { container } = render(
      <ThemeProvider theme={lightTheme}>
        <InputByRef id="input" label="Input Label" />
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should call onChange when the input is changed', () => {
    const handleChange = jest.fn();
    const { getByLabelText } = render(
      <ThemeProvider theme={lightTheme}>
        <InputByRef id="input" label="Input Label" onChange={handleChange} />
      </ThemeProvider>,
    );

    waitFor(() =>
      fireEvent.change(getByLabelText('Input Label') as HTMLInputElement, {
        target: { value: 'new value' },
      }),
    );
    waitFor(() => expect(handleChange).toHaveBeenCalled());
  });

  it('should have an error state when isError prop is true', () => {
    const { getByLabelText, getByText } = render(
      <ThemeProvider theme={lightTheme}>
        <InputByRef id="input" label="Input Label" isError error="Input is required" />
      </ThemeProvider>,
    );

    const errorMessage = getByText('Input is required');
    waitFor(() => expect(getByLabelText('Input Label').classList.contains('error')).toBe(true));
    waitFor(() => expect(errorMessage).toBeInTheDocument());
  });
});

describe('PhoneInput', () => {
  it('renders a label and input element', () => {
    render(
      <ThemeProvider theme={lightTheme}>
        <PhoneInput label="Phone Number" />
      </ThemeProvider>,
    );
    waitFor(() => expect(screen.getByText(/phone number/i)).toBeInTheDocument());
    waitFor(() => expect(screen.getByRole('textbox')).toBeInTheDocument());
  });

  it('displays an error message when passed an error prop', () => {
    const errorMessage = 'Please enter a valid phone number';
    render(
      <ThemeProvider theme={lightTheme}>
        <PhoneInput label="Phone Number" error={errorMessage} />
      </ThemeProvider>,
    );
    waitFor(() => expect(screen.getByText(errorMessage)).toBeInTheDocument());
  });

  it('applies the disabled style when passed the disabled prop', () => {
    render(
      <ThemeProvider theme={lightTheme}>
        <PhoneInput label="Phone Number" disabled />
      </ThemeProvider>,
    );
    waitFor(() => expect(screen.getByRole('textbox')).toBeDisabled());
    waitFor(() => expect(screen.getByText(/phone number/i)).toHaveStyle({ color: '#B1B5B9' }));
  });
});

describe('EmailInput', () => {
  const props = {
    id: 'email-input',
    label: 'Email',
    placeholder: 'Enter your email address',
    hint: 'We will never share your email with anyone.',
  };

  it('renders without crashing', () => {
    const { getByLabelText } = render(
      <ThemeProvider theme={lightTheme}>
        <EmailInput {...props} />
      </ThemeProvider>,
    );
    waitFor(() => expect(getByLabelText(props.label)).toBeInTheDocument());
  });
  it('shows the label when the input is focused', () => {
    const { getByLabelText } = render(
      <ThemeProvider theme={lightTheme}>
        <EmailInput {...props} />
      </ThemeProvider>,
    );

    waitFor(() => fireEvent.focus(getByLabelText(props.label)));

    waitFor(() => expect(screen.getByText(props.label)).toHaveStyle('opacity: 1'));
  });
  it('shows the hint when the input is focused', () => {
    const { getByLabelText } = render(
      <ThemeProvider theme={lightTheme}>
        <EmailInput {...props} />
      </ThemeProvider>,
    );
    waitFor(() => fireEvent.focus(getByLabelText(props.label)));
    waitFor(() => expect(screen.getByText(props.hint)).toBeInTheDocument());
  });
});

describe('PasswordInput', () => {
  it('renders with label and placeholder', () => {
    const { getByLabelText, getByPlaceholderText } = render(
      <ThemeProvider theme={lightTheme}>
        <PasswordInput id="test" label="Test" placeholder="Test Placeholder" />
      </ThemeProvider>,
    );
    waitFor(() => expect(getByLabelText('Test')).toBeInTheDocument());
    waitFor(() => expect(getByPlaceholderText('Test Placeholder')).toBeInTheDocument());
  });

  it('shows password on click', () => {
    const { getByLabelText, getByTestId } = render(
      <ThemeProvider theme={lightTheme}>
        <PasswordInput id="test" label="Test" />
      </ThemeProvider>,
    );
    const passwordInput = waitFor(() => getByLabelText('Test') as HTMLInputElement);

    waitFor(() => expect(getByLabelText('Test')).toBe('password'));
    waitFor(() => fireEvent.click(getByTestId('show-hide-button')));
    waitFor(() => expect(passwordInput).toBe('text'));
    waitFor(() => fireEvent.click(getByTestId('show-hide-button')));

    waitFor(() => expect(passwordInput).toBe('password'));
  });
});
