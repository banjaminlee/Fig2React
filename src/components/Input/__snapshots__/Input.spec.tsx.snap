// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputByRef should render correctly 1`] = `
<div
  class="sc-dkrFOg sc-eDvSVe esOpsw dABIzc"
  color="text_primary"
>
  Input Label
</div>
`;
